### ISSUE TITLE
backupccl: backup/restore does not handle comments

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#44396]
Since 19.2 CockroachDB supports the COMMENT statement to decorate database, tables and columns. We plan to extend it for sequences, views, indexes as well (#44135).

Comments are stored in a separate table `system.comments`.

`cockroach dump` in the CLI handles this by emitting the `COMMENT` statements alongside `CREATE` in the output.

However BACKUP does not capture the comments and they are thus not restorable.

Jira issue: CRDB-5231

**Comment by knz** [comment]
cc @hueypark FYI

**Comment by knz** [comment]
@jseldess do you still use the "docs-known-limitation" label? Otherwise what's the current process to report and document known limitations?

**Comment by dt** [comment]
On the bright side, I think the new "full-cluster backup" in 20.1 _will_ preserve comments just fine, but backing up individual tables or databases currently will not.

**Comment by jseldess** [comment]
@knz, yes, we use that label to identify new known limitations to document for a GA release. Once we have in the docs, we add `docs-done`.

**Comment by kenliu** [comment]
Nice catch @knz

**Comment by knz** [comment]
@dt caught it. I merely filed the issue.

**Comment by jseldess** [comment]
Documented in https://github.com/cockroachdb/docs/pull/7118.

**Comment by hueypark** [comment]
@knz I think we already resolved this issue(#43152) and merged it to [20.1 branch](https://github.com/cockroachdb/cockroach/tree/release-20.1). Am I wrong?

**Comment by knz** [comment]
yes, correct. thanks for noticing!

**Comment by hueypark** [comment]
@jseldess We should also reflect on the changes in the [20.1 documentation](https://github.com/cockroachdb/docs/blob/master/v20.1/known-limitations.md).

**Comment by pbardea** [comment]
Hm, unfortunately I don't think that #43152 quite resolves this issue. A BACKUP of a table or database will not by default include `system.comments`. This table would need to be backed up and manually restored and re-inserted (by first restoring it into a temporary db). So upon restoring there is no way to retrieve the comments.

**Comment by hueypark** [comment]
Interesting. Let me look a little further.

**Comment by adityamaru** [comment]
We are considering changing the format we store backup metadata as a project for our 21.2 release. This new format would unlock the ability to store larger types of metadata egs: comments, stats and so we should off on addressing this issue until then!

**Comment by hand-crdb** [comment]
A demo of a workaround is at [https://github.com/hand-crdb/preserve_comments_cockroachdb](https://github.com/hand-crdb/preserve_comments_cockroachdb)

**Comment by dt** [comment]
(claude) Done: `system.comments` is backed up and restored, with coverage in `pkg/backup/full_cluster_backup_restore_test.go` and `pkg/backup/backup_test.go`.
